data/jobfactory.py

```diff
@@ -81,19 +81,22 @@ def __init__(self, user, workflow_version, stage_group, user_job_order, system_j
         self.share_group = share_group
         self.fund_code = fund_code
 
-    def create_job(self):
+    def create_job(self, is_authorized):
```

Comment:
Why pass the `is_authorized` variable from api.py instead of reading `settings.REQUIRE_JOB_TOKEN` directly inside this function? My perspective on reading the code is that this function is the only place that _cares_ about that setting, and it's not an API-level concern, so we don't need to complicate the code in `api.py` or the interface to `create_job()`

Regarding the variable name, I tend to think of the `is_` prefix being for functions (it reads as a question) rather than variable names. Also, it might be good to provide a default value of `False`.